# Hi, I'm Amirtesh Raghuram 

I have a strong interest in **Machine Learning**, **Deep Learning**, and **Bioinformatics**. I am currently in my second year, pursuing a B.Tech in Biotechnology at VIT, Vellore, where I am developing my computational and bioinformatics skills actively. 

## **Current Skills**

### **Data Science and Machine Learning**  
#### Python:  
- Proficient in **Scikit-learn**, **XGBoost**, **LightGBM**, and **CatBoost** for building predictive models.  
- Proficient in implementing **Deep Learning** techniques using **PyTorch** for tabular and image data, with experience in building neural networks.  
- Basics of **TensorFlow** for deep learning tasks.  
- Experience with **data manipulation** and **visualization** libraries, including:  
  - **NumPy** and **Pandas** for data manipulation and analysis.  
  - **Matplotlib** and **Seaborn** for creating insightful visualizations.  
- Developed a library called [**Torchify**](https://github.com/Amirtesh/Pytorch-Torchify) to simplify PyTorch workflows with features like **model compilation**, **training (fit)**, **predictions**, and performance evaluation.  

#### R:  
- Basics of data analysis and visualization using **dplyr** and **ggplot2**.  
- Basics of machine learning workflows in R.  

---

### **Computational Drug Discovery**  
- Performing **Protein-Ligand Docking** using tools like **AutoDock Vina**, **Smina**, **QVina**, **SwissDock**, **AutoDock 4.2**, and **ADFR suite** for both rigid and flexible docking.  
- Performing **Molecular Dynamics Simulations** with **GROMACS**.  
- Performing **MMPBSA analysis** using **gmx_MMPBSA** to calculate binding free energies.  
- Using molecular visualization and modification tools such as **PyMol**, **Chimera**, **Python Molecular Viewer (PMV)**, **AutoDock Tools**, and **Discovery Studio Visualizer**.  
- Proficient in **RDKit** for cheminformatics, including molecular descriptor calculation, fingerprint generation, and chemical property prediction.  
- Utilizing computational tools for drug discovery, including **SwissADME**, **Swiss Target Prediction**, **Swiss Similarity**, **Protox**, **MolSoft**, and **pkcsm**.  

---

### **BioPython**  
- Proficient in using **BioPython** for bioinformatics tasks, including:  
  - Reading and processing sequence files in formats like **FASTA** and **GenBank**.  
  - Creating and manipulating **SeqRecord** and **SeqFeature** objects.  
  - Performing sequence alignment using tools like **ClustalW**, **MUSCLE**, and **EMBOSS Needle/Water** (both within Python and via EMBL web tools).  
  - Automating **BLAST** for sequence similarity searches, running **blastn**, **blastp**, **blastx**, and **tblastn**.  
  - Working with phylogenetic data using **Bio.Phylo**.  
  - Writing output files in various biological formats.  
  - Familiar with **pairwise sequence alignment** for similarity studies.  

---

### **RNA-Seq Analysis**  
- Performing **quality checks** on FASTQ files using **fastqc**.  
- **Trimming raw reads** with **Trimmomatic**.  
- Aligning sequences to reference genomes using **HISAT2**.  
- Building a **feature count matrix** using the **featureCounts** tool from **Subread**.  
- Automating RNA-seq workflows for multiple FASTQ files to create combined counts data and metadata using **bash scripts**.  

  

### **Let’s Connect**  
Feel free to reach out if you’d like to discuss projects, research, or opportunities!  
[LinkedIn](https://in.linkedin.com/in/amirtesh-raghuram-90161828a)

**Email**- amirtesh21.5@gmail.com
