# Hi, I'm Amirtesh Raghuram 

I have a strong interest in **Machine Learning**, **Deep Learning**, and **Bioinformatics**. I am currently in my second year, pursuing a B.Tech in Biotechnology at VIT, Vellore, where I am developing my computational and bioinformatics skills actively. 

## Current Skills:

### Machine Learning:
- Proficient in **Scikit-learn**, **XGBoost**, **LightGBM**, and **CatBoost** for building predictive models.
- Proficient in implementing **Deep Learning** techniques using **PyTorch** and building neural networks to work on tabular and image data, along with basics of Tensorflow.
- I have also created a simple library called [Torchify](https://github.com/Amirtesh/Pytorch-Torchify) which can be used to quickly train models built with **PyTorch** with various simple methods like **compile**, **fit**, **predict** and quickly evaluating performance of the model on a dataset with **performance** method. It only requires the building of model and removes the boilerplate code for training it and also can include various features such as **learning rate scheduling**, **gradient clipping** and plotting of essential metrics to see the training progress of the model.

### **Bioinformatics:**
- Performing **Protein-Ligand Docking** using tools like **AutoDock Vina**, **Smina**, **QVina**, **SwissDock**, **AutoDock 4.2**, and **ADFR suite** for both rigid and flexible docking.  
- Performing **Molecular Dynamics Simulations** with **GROMACS**.  
- Performing **MMPBSA analysis** using **gmx_MMPBSA** to calculate binding free energies.  
- Usage of molecular visualization and modification tools such as **PyMol**, **Chimera**, **Python Molecular Viewer (PMV)**, **AutoDock Tools**, and **Discovery Studio Visualizer**.  
- Usage of **computational drug discovery tools** such as **SwissADME**, **Swiss Target Prediction**, **Swiss Similarity**, **Protox**, **MolSoft**, **pkcsm**, and more.  
- Usage of **RDKit** in Python for working with molecules.  
- **BioPython**:
  - Reading and processing biological sequence files in formats such as **FASTA** and **GenBank**.  
  - Creating and manipulating **SeqRecord** and **SeqFeature** objects.  
  - Reading in sequence alignment files.  
  - Performing sequence alignment using tools like **ClustalW**, **MUSCLE**, and **EMBOSS Needle/Water** (both within Python and via the EMBL web tools).  
  - Writing output files in various biological formats.  
  - Familiar with **pairwise sequence alignment** for sequence similarity studies.  
  - Familiar with **BLAST** for sequence similarity searches and running **blastn**, **blastp**, **blastx**, and **tblastn** using both the **NCBI browser** and **Biopython**.  
  - Basics of using **Bio.Phylo** for working with phylogenetic data.  
- **RNA-Seq Analysis**:
  - Performing **quality checks** on FASTQ files using tools like **FastQC**.  
  - **Trimming raw reads** with **Trimmomatic**.  
  - Aligning sequences to reference genomes using **HISAT2**.  
  - Building a **feature count matrix** using the **featureCounts** tool from Subread.  
  - Automating RNA-seq workflows for multiple FASTQ files using **bash scripts**.  




### R Programming:
- Basics of data analysis and visualization libraries such as dplyr and ggplot2 in R.
- Basics of machine learning with R.
  

### **Let’s Connect**  
Feel free to reach out if you’d like to discuss projects, research, or opportunities!  
[LinkedIn](https://in.linkedin.com/in/amirtesh-raghuram-90161828a)

**Email**- amirtesh21.5@gmail.com
